pkg/testcase/certificates.go

```diff
@@ -0,0 +1,122 @@
+package testcase
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/shared"
+)
+
+func errCheck(error error, message string, ignore bool) {
+	if error != nil {
+		shared.LogLevel("error", message)
+		if !ignore {
+			Expect(error).NotTo(HaveOccurred())
+		}
+
+	}
+}
+
+// Rotate certificate for etcd only and cp only nodes
+func certRotate(product string, ip string) {
+	// Stop service on server
+	_, stopError := shared.StopService(product, ip, "server")
+	errCheck(stopError, fmt.Sprintf("Error stopping %s service", product), false)
+	time.Sleep(time.Duration(10) * time.Second) // Adding 10 second sleep between steps
+	// Rotate certificate
+	_, rotateError := shared.CertRotate(product, ip)
+	errCheck(rotateError, fmt.Sprintf("Error running certificate rotate for %s service", product), false)
+	time.Sleep(time.Duration(10) * time.Second) // Adding 10 second sleep between steps
+	// start service on server
+	_, startError := shared.StartService(product, ip, "server")
+	errCheck(startError, fmt.Sprintf("Error starting %s service", product), true)
+	time.Sleep(time.Duration(10) * time.Second) // Adding 10 second sleep between steps
+	// Retry Stop start of service. (Inconsistent failure to start service above,
+	// can be fixed hopefully with a restart)
+	if startError != nil {
+		_, retryError := shared.StopStartService(product, ip, "server")
+		errCheck(retryError, fmt.Sprintf("ERROR: Retrying Stop/Start %s service", product), false)
+	}
+}
+
+// Compare TLS Directories before and after cert rotation to display identical files
+func compareTLSDir(product string, ip string) (string, error) {
+	dataDir := fmt.Sprintf("/var/lib/rancher/%s", product)
+	serverDir := fmt.Sprintf("%s/server", dataDir)
+	origTLSDir := fmt.Sprintf("%s/tls", serverDir)
+	cmd := fmt.Sprintf("sudo ls -lt %s/ | grep tls | awk {'print $9'} | sed -n '2 p'", serverDir)
+	tlsDir, error := shared.RunCommandOnNode(cmd, ip)
+	if error != nil {
+		shared.LogLevel("warn", "Unable to get new TLS Directory name")
+	}
+	shared.LogLevel("info", fmt.Sprintf("TLS Directory name: %s", tlsDir))
+	newTLSDir := fmt.Sprintf("%s/%s", serverDir, tlsDir)
+	shared.LogLevel("info", "Comparing Directories: %s and %s", origTLSDir, newTLSDir)
+	cmd2 := fmt.Sprintf("sudo diff -sr %s/ %s/ | grep -i identical | "+
+		"awk '{print $2}' | xargs basename -a | "+
+		"awk 'BEGIN{print \"Identical Files:  \"}; {print $1}'", origTLSDir, newTLSDir)
+	return shared.RunCommandOnNode(cmd2, ip)
+}
+
+func TestCertRotate() {
```

Comment:
your first function of the file should be this one. the exported

Reply:
fixed